bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/LedgerType.java

```diff
@@ -0,0 +1,46 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.bookkeeper.client.api;
+
+/**
+ * Describes the type of ledger.
+ * LedgerTypes describes the behaviour of the ledger in respect to durability and provides
+ * hints to the storage of data on Bookies
+ *
+ * @since 4.6
+ */
+public enum LedgerType {
+    /**
+     * Persistent Durability, using Journal.<br>
+     * Each entry is persisted to the journal and every writes receives and acknowledgement only with the guarantee that
+     * it has been persisted durabily to it (data is fsync'd to the disk)
+     */
+    PD_JOURNAL,
```

Comment:
These names aren't very descriptive. I think something like FSYNC_REQUIRED and FSYNC_DEFERRED would be clearer.

Reply:
The more I am working on this the more I am convinced that these names are not fine.

We have to use the word 'journal" because @jvrao is working on a LedgerType without the Journal.
And Volatile is the opposite of Durability

Maybe
SYNC_REQUIRED_WITH_JOURNAL
SYNC_EXPLICIT_WITH_JOURNAL

@sijie @jvrao  ?

Reply:
I am fine with REQUIRED and DEFERRED.

Reply:
sorry, @sijie 

SYNC_REQUIRED_WITH_JOURNAL, SYNC_DEFERRED_WITH_JOURNAL
or
FSYNC_REQUIRED, FSYNC_DEFERRED ?

Reply:
FSYNC_REQUIRED_WITH_JOURNAL, FSYNC_DEFERRED_WITH_JOURNAL ?

Reply:
FSYNC_REQUIRED_ON_JOURNAL, FSYNC_DEFERRED_ON_JOURNAL ?

Reply:
Better 'with' as some ledger type won't have journal

Reply:
Wouldn't these have a different type though, something like SKIP_JOURNAL?

Reply:
@ivankelly this is per ledger, allowing a storage implementation bypassing journal.

Reply:
If we rename 'sync' to force, for the LedgerType @sijie  @ivankelly what is your idea ?
FSYNC_REQUIRED_ON_JOURNAL vs FORCE_REQUIRED_ON_JOURNAL vs AUTOMATIC_FORCE_ON_JOURNAL
and
FSYNC_DEFERRED_ON_JOURNAL vs FORCE_DEFERRED_ON_JOURNAL vs EXPLICIT_FORCE_ON_JOURNAL

and SKIP_JOURNAL (@jvrao)

Reply:
FORCE_ON_JOURNAL (default)
FORCE_DEFERRED_ON_JOURNAL
SKIP_JOURNAL

Reply:
@ivankelly  @sijie  @jvrao 
Proposal from Ivan sounds very good to me

FORCE_ON_JOURNAL (default)
FORCE_DEFERRED_ON_JOURNAL

SKIP_JOURNAL (this is a hint for the future, not part of BP-14)

Reply:
sgtm